vip-cache-manager/includes/class-rest-api.php

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * Plugin's REST API
+ *
+ * @package a8c_Cache_Manager
+ */
+
+require_once( __DIR__ . '/class-singleton.php' );
+
+/**
+ * REST API class
+ */
+class REST_API extends Singleton {
+
+	/**
+	 * API SETUP
+	 */
+	const API_NAMESPACE  = 'cache-manager/v1';
+	const ENDPOINT_PURGE = 'purge'; // purge URL(s)
+	const ENDPOINT_BAN   = 'ban'; // ban URL(s)
+	const URLS_KEY       = 'urls';
+
+	/**
+	 * PLUGIN SETUP
+	 */
+
+	/**
+	 * Register hooks
+	 */
+	protected function class_init() {
+		add_action( 'rest_api_init', array( $this, 'rest_api_init' ) );
+	}
+
+	/**
+	 * PLUGIN FUNCTIONALITY
+	 */
+
+	/**
+	 * Register API routes
+	 */
+	public function rest_api_init() {
+
+		register_rest_route( self::API_NAMESPACE, '/' . self::ENDPOINT_PURGE, array(
+			'methods'             => 'POST',
+			'callback'            => array( $this, 'purge_urls' ),
+			'permission_callback' => array( $this, 'check_secret' ),
```

Comment:
As mentioned in Slack, it would be best to reuse `wpcom_vip_go_rest_api_request_allowed()` here. Since it uses a header, it's slightly easier to obscure in logs as necessary.

Reply:
Sure, will work on this, makes definitely sense.